nano/node/messages.cpp

```diff
@@ -940,6 +940,21 @@ std::size_t nano::confirm_req::size (nano::block_type type_a, std::size_t count)
 	return result;
 }
 
+std::string nano::confirm_req::to_string () const
```

Comment:
confirm_req has to modes of operation:
* list of hash pair
* block to confirm
You are only handling the list of pairs mode.
You need to handle both modes.